src/compiler/core.ts

```diff
@@ -1773,9 +1773,27 @@ namespace ts {
         if (a === b) return Comparison.EqualTo;
         if (a === undefined) return Comparison.LessThan;
         if (b === undefined) return Comparison.GreaterThan;
-        a = a.toUpperCase();
-        b = b.toUpperCase();
-        return a < b ? Comparison.LessThan : a > b ? Comparison.GreaterThan : Comparison.EqualTo;
+        const aLength = a.length;
+        const bLength = b.length;
+        const maxLength = Math.min(aLength, bLength);

```

Comment:
"minLength"?

Reply:
Capping values always has this unintuitive aspect. The max index of two strings is always determined by the min of the two lengths. Similarly, if you needed to cap a potentially-negative index to `0`, you'd call it a `minIndex`, but call `Math.max`.

Reply:
"maxIndex" would be fine, but it's the smaller (i.e. minimum) of the two lengths.

Reply:
Well...it's not a valid index 😄

`lengthUpperBound`?

Reply:
`commonLength`?